controlplane/pkg/nsmd/nsmd_crossconnect_client.go

```diff
@@ -452,14 +454,16 @@ func (client *NsmMonitorCrossConnectClient) remotePeerConnectionMonitor(ctx cont
 
 	if err != nil {
 		span.LogError(err)
-		// Same as DST down case, we need to wait for same NSE and updated NSMD.
 		connections := client.xconManager.GetClientConnectionByRemote(remotePeer)
 		for _, cc := range connections {
 			ccSpan := common.SpanHelperFromConnection(ctx, cc, "remotePeerConnectionMonitor.update")
 			defer ccSpan.Finish()
 			ccSpan.LogObject("clientConnection", cc)
 			client.xconManager.DestinationDown(ccSpan.Context(), cc, true)
 		}
+		client.remotePeerLock.Lock()
```

Comment:
It seems we need to update this map properly and not here, please check why it was not empty.

Reply:
yes, you are right. The problem was happening because monitoring goroutine sometimes closes (for example stream returned error) and the listener `AddCliendtConnection` not checks that the goroutine already finished.

Scenario:
1. Peer monitoring goroutine finishes by received error 
2. Model receives new client connection
3. Method `AddCliendtConnection` looks into a map of peers and see that peer already exists and exits.

This PR  fixes this scenario.